src/spikeinterface/core/baserecordingsnippets.py

```diff
@@ -357,7 +357,12 @@ def get_channel_locations(self, channel_ids=None, axes: str = "xy"):
                 # check that multiple probes are non-overlapping
                 check_probe_do_not_overlap(all_probes)
                 all_positions = np.vstack([probe.contact_positions for probe in all_probes])
-                positions = all_positions[channel_indices]
+                probes_channel_indices = np.concatenate([probe.device_channel_indices for probe in all_probes])
+
+                sorted_probe_idx = np.argsort(probes_channel_indices)
+                sorted_positions_idx = np.searchsorted(probes_channel_indices[sorted_probe_idx], channel_indices)
+
+                positions = all_positions[sorted_probe_idx[sorted_positions_idx]]
```

Comment:
OK for me but this is really strange because when attached to a recording the device_channel_indices should always be sorted and should be =[0...n-1].
Maybe there is something deeper in the bug.